### PR TITLE
create more classes in separated files and set up associations

### PR DESCRIPTION
**Pull request by huzaifakhan0308**
Set up associations between classes and objects.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rentals(date)
+    rental = Rental.new(date)
+    @rentals << rental
+    rental.book = self
+    rental
```
Comment (mckent05) [inline book.rb]: - You have not properly implemented the `has-many` association between the `book` and `rental` class. Please fix this 🙏 . I expect you would have something like this:
```ruby
def add_rentals(date, person)
  Rental.new(date, self, person)
end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
```
Comment (mckent05) [inline rental.rb:7]: - You are yet to implement the `belongs-to` relationship between the `rental` class and the `book` and `person` class. i.e 
```ruby
book.rentals << self
```
```ruby
person.rentals << self
```

File: student.rb [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
   end
 
   def play_hooky
     '¯\\(ツ)/¯'
   end
+
+  def classrooms(classroom)
+    @classroom = classroom
+    classroom.add_student(self)
+  end
```
Comment (mckent05) [inline student.rb]: - Please implement a custom setter that serves as a `belongs-to` relationship etween the `student` class and the `classroom` class. i.e
 ```ruby
    def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: student.rb [hunk 4]
```diff
@@ -1,12 +1,19 @@
+require_relative 'classroom'
+
 class Student < Person
   attr_accessor :classroom
 
-  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+  def initialize(age, name = 'Unknown', parent_permission: true)
     super(name, age, parent_permission)
-    @classroom = classroom
+    @classroom = nil
```
Comment (mckent05) [inline student.rb]: -I suggest initializing this classroom variable like this `@classroom=classroom`. Please add the `classroom` variable as your instance variable 🙏

File: person.rb [hunk 5]
```diff
@@ -29,4 +32,9 @@ def of_age?
 
     false
   end
+
+  def add_rentals(date)
+    rental = Rental.new(date)
+    @rentals << rental
+  end
```
Comment (mckent05) [inline person.rb]: - Similar to my suggestion for the `book` class, please implement properly the `has-many` relationship between the `person` class and the `rentals` class 🙏

File: classroom.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+require_relative 'student'
+
+class Classroom
+  attr_accessor :label, :students
```
Comment (mckent05) [inline classroom.rb]: - The `students` variable should only have a getter method. Please fix this 🙏

File: classroom.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+    student.classrooms(self)
```
Comment (mckent05) [inline classroom.rb]: - According to the project requirements, you have not properly assigned the classroom to the student, please fix this 🙏 
it should be `student.classrom = self`.

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @huzaifakhan0308, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job creating associations between your classes 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @huzaifakhan0308, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**